cmake/Dependencies.cmake

```diff
@@ -1828,10 +1828,25 @@ endif()
 if(USE_KINETO)
   if((NOT USE_CUDA) OR MSVC)
     set(LIBKINETO_NOCUPTI ON CACHE STRING "")
+  endif()
+  if(NOT USE_ROCM)
+    set(LIBKINETO_NOROCTRACER ON CACHE STRING "")
+  endif()
+
+  if(LIBKINETO_NOCUPTI AND LIBKINETO_NOROCTRACER)
     message(STATUS "Using CPU-only version of Kineto")
   else()
-    set(LIBKINETO_NOCUPTI OFF CACHE STRING "")
-    message(STATUS "Using Kineto with CUPTI support")
+    if(USE_ROCM)
+      if(NOT ENV{ROCM_SOURCE_DIR})
+        set(ENV{ROCM_SOURCE_DIR} "/opt/rocm")
+      endif()
+      set(LIBKINETO_NOROCTRACER OFF CACHE STRING "")
+      message(STATUS "Using Kineto with Roctracer support")
+    endif()
+    if(USE_CUDA)
+      set(LIBKINETO_NOCUPTI OFF CACHE STRING "")
+      message(STATUS "Using Kineto with CUPTI support")
+    endif()
```

Comment:
This part seems wrong if `MSVC` is set to true and `USE_CUDA` is set to true

Reply:
Sorry, it is a bit convoluted.  I tried to make minimal changes to something that clearly only considers one tracing library.
There is also a hidden relation: USE_CUDA && USE_ROCM == False.  These never coexist.

So the 1846 guard can only be reached if 1829 is not entered.  If MSVC is set then LIBKINETO_NOCUPTI is ON.

At that point the only way into 1838 is if we are a ROCm build.  So then the 1846 is always false.

Also, there is no rocm on windows (i.e. msvc).

Happy to refactor it.  Suggestions welcome.

Reply:
Hmm I guess here it is a bit confusing to me because 
both `LIBKINETO_NOCUPTI` and `LIBKINETO_NOROCTRACER` are cached strings. therefore users can override it. and the set statement here are noops if user has explicitly set cache.